lib/logstash/inputs/syslog.rb

```diff
@@ -165,7 +184,7 @@ def tcp_listener(output_queue)
 
   public
   def teardown
-    @shutdown_requested = true
+    @shutdown_requested.update{|v| true}
```

Comment:
meh, personally I am not a fan of extra spaces for `{ }`... I prefer without, just like parenthesis.